spec.html

```diff
@@ -9989,6 +9989,72 @@ <h1>The Environment Record Type Hierarchy</h1>
       <emu-clause id="sec-declarative-environment-records">
         <h1>Declarative Environment Records</h1>
         <p>Each <dfn variants="Declarative Environment Records">Declarative Environment Record</dfn> is associated with an ECMAScript program scope containing variable, constant, let, class, module, import, and/or function declarations. A Declarative Environment Record binds the set of identifiers defined by the declarations contained within its scope.</p>
+        <p>Declarative Environment Records have the additional fields listed in <emu-xref href="#table-additional-fields-of-declarative-environment-records"></emu-xref>.</p>
+        <emu-table id="table-additional-fields-of-declarative-environment-records" caption="Additional Fields of Declarative Environment Records">
+          <table>
+            <tr>
+              <th>
+                Field Name
+              </th>
+              <th>
+                Value
+              </th>
+              <th>
+                Meaning
+              </th>
+            </tr>
+            <tr>
+              <td>
+                [[Bindings]]
+              </td>
+              <td>
+                a List of DeclarativeBindings
+              </td>
+              <td>
+                Satisfies the invariant that no two records in [[Bindings]] have the same [[BoundName]].
+              </td>
+            </tr>
+          </table>
+        </emu-table>
+
+        <p>A <dfn>DeclarativeBinding</dfn> is either a SimpleDeclarativeBinding or an ImportDeclarativeBinding. SimpleDeclarativeBindings can appear in any Declarative Environment Record, but ImportDeclarativeBindings can only appear in a Module Environment Record.</p>
+
+        <p>A <dfn>SimpleDeclarativeBinding</dfn> has the following fields:</p>
+        <emu-table id="table-fields-of-simpledeclarativebindings" caption="SimpleDeclarativeBinding Fields">
+          <table>
+            <tr>
+              <th>Field Name</th>
+              <th>Value</th>
+              <th>Meaning</th>
+            </tr>
+            <tr>
+              <td>[[BoundName]]</td>
```

Comment:
This field needs to be moved to the description of `DeclarativeBinding` since we refer to it on values that are known only to be a `DeclarativeBinding`.

Reply:
Are you saying it needs to be that way in order for the spec to be "correct" in some sense? Because it sounds like you're talking about statically typing the pseudocode, and you have in mind a specific rule for how that should work.

As a counter-example, consider `ComposeWriteEventBytes`, which has `_W_.[[ByteIndex]]`, where `_W_` is either a WriteSharedMemory event or a ReadModifyWriteSharedMemory event, each of which separately declares a `[[ByteIndex]]` field. There are several other such examples.